app/models/answer.rb

```diff
@@ -368,4 +369,24 @@ def reset_location_flag
     self.location_values_replicated = false
     true
   end
+
+  def validate_date
+    raw_date = read_attribute_before_type_cast("date_value")
+
+    begin
+      raw_date.present? && raw_date.is_a?(String) && Date.parse(raw_date)
+    rescue ActiveRecord::RecordInvalid
```

Comment:
Another thought: instead of duplicating Rails' date or datetime parsing stuff in this validator, maybe we could do:

```
return unless read_attribute_before_type_cast(:date_value).present? && date_value.nil?
errors.add(:date_value, :invalid_date)
```

The reasoning being that if the attrib is set before typecast but then it turns into nil when it's typecast, it mustn't have been valid. This way we rely on Rails to do the heavy lifting of checking if the date is valid or now.

It's also much less code. :)

Reply:
In any case though I think we do need a test for the validation code. (Not a feature spec, just a model spec). So try saving an answer with a valid date and with a junk date, ensure correct valid/invalid for both, do the same with datetime. See the bottom of the skip_rule_spec.rb file for an example. This should be easy, only 15-20 minutes of work. If it's taking much more than that please ask for help.